### ISSUE TITLE
Allow merging of secrets into .yml config

### ISSUE THREAD

**Issue by simonmittag** [simonmittag/j8a#197]
**Is your feature request related to a problem? Please describe.**
Currently, the .yml config file is static and does not have a way to embed a secret. Secrets, i.e. private keys and certificates have to be passed alongside the entire config. This makes working with an external secret store difficult 

**Workarounds**
***Envsubst***
Pass the secret to an environment variable and merge into config file via envsubst before starting the server:  https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html

**Describe the solution you'd like**
Helm, Harness and similar tools use go templating (https://pkg.go.dev/text/template) to merge values into .yml configuration files. I suggest we change j8a’s config parsing to allow for this after reading the configuration. Currently, given the config can be loaded either from disk or ENV variable itself, that will have to happen after loading, but before converting the configuration to JSON which we use internally to map to a struct.

See: https://github.com/simonmittag/j8a/blob/master/config.go#L37-L51
```
func (config Config) load() *Config {
	if len(ConfigFile) > 0 {
		log.Debug().Msgf("attempting config from file '%s'", ConfigFile)
		if cfg := *config.readYmlFile(ConfigFile); cfg.ok() {
			config = cfg
		} else {
			config.panic(fmt.Sprintf("error loading config from file '%s'", ConfigFile))
		}
	} else if len(os.Getenv(J8ACFG_YML)) > 0 {
		log.Debug().Msgf("attempting config from env %s", J8ACFG_YML)
		if cfg := *config.readYmlEnv(); cfg.ok() {
			config = cfg
		} else {
			config.panic(fmt.Sprintf("error loading config from env '%s'", J8ACFG_YML))
		}
}
```

and how it is called with command like pattern during server Bootstrap https://github.com/simonmittag/j8a/blob/master/server.go#L106-L121

```
config := new(Config).
		load().
		reApplyResourceSchemes().
		reApplyResourceNames().
		validateJwt().
		compileRoutePaths().
		compileRouteTransforms().
		validateRoutes().
		addDefaultPolicy().
		setDefaultUpstreamParams().
		setDefaultDownstreamParams().
		validateHTTPConfig().
		validateAcmeConfig()
```

All the validations should still function after the templating is enabled.

**Comment by simonmittag** [comment]
cc @petelah

**Comment by petelah** [comment]
I think our best option is to allow yaml templating.

Since we already validate the yaml by parsing. Before we validate we can run the yaml file through golang's templating library: [https://pkg.go.dev/html/template](https://pkg.go.dev/html/template) and fill in the templated variables `{{.Var}}` with the corresponding environment variable.

I think it is a good idea to not incorporate other secret management tools and instead rely on the user to supply them on their own and store them as environment variables, this makes things like injecting secrets via tools like terraform, terraform, etc more compatible overall since they all have their own secret management and injection taken care of out of the box.

**Comment by petelah** [comment]
To further this option. If you look at yaml files used with helm they all use go-template to inject variables into yaml config files.

[Helm Docs](https://helm.sh/docs/chart_template_guide/).

**Comment by simonmittag** [comment]
I’m supportive of this idea after looking at how others, including helm and harness solved this problem here: https://docs.harness.io/article/mwy6zgz8gu-use-go-templating-in-kubernetes-manifests so I’m upgrading this ticket to an enhancement

**Comment by simonmittag** [comment]
@petelah bumping this, i want to make a 0.9.1 release with several smaller fixes, including this one.
